enforcer/src/main/java/org/wso2/micro/gateway/enforcer/security/jwt/issuer/HttpTokenServerHandler.java

```diff
@@ -0,0 +1,137 @@
+import io.grpc.netty.shaded.io.netty.channel.ChannelFutureListener;
+import io.grpc.netty.shaded.io.netty.channel.ChannelHandlerContext;
+import io.grpc.netty.shaded.io.netty.channel.SimpleChannelInboundHandler;
+import io.grpc.netty.shaded.io.netty.handler.codec.http.DefaultFullHttpResponse;
+import io.grpc.netty.shaded.io.netty.handler.codec.http.FullHttpResponse;
+import io.grpc.netty.shaded.io.netty.handler.codec.http.HttpObject;
+import io.grpc.netty.shaded.io.netty.handler.codec.http.HttpRequest;
+import io.grpc.netty.shaded.io.netty.handler.codec.http.HttpUtil;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.wso2.micro.gateway.enforcer.config.ConfigHolder;
+import org.wso2.micro.gateway.enforcer.config.dto.CredentialDto;
+import org.wso2.micro.gateway.enforcer.constants.APIConstants;
+import org.wso2.micro.gateway.enforcer.dto.APIKeyValidationInfoDTO;
+import org.wso2.micro.gateway.enforcer.security.TokenValidationContext;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+
+import static io.grpc.netty.shaded.io.netty.handler.codec.http.HttpHeaderNames.CONTENT_LENGTH;
+import static io.grpc.netty.shaded.io.netty.handler.codec.http.HttpHeaderValues.CLOSE;
+import static io.grpc.netty.shaded.io.netty.handler.codec.http.HttpHeaderValues.KEEP_ALIVE;
+import static io.grpc.netty.shaded.io.netty.handler.codec.http.HttpHeaderValues.TEXT_PLAIN;
+import static io.grpc.netty.shaded.io.netty.handler.codec.http.HttpHeaders.Names.CONNECTION;
+import static io.grpc.netty.shaded.io.netty.handler.codec.http.HttpHeaders.Names.CONTENT_TYPE;
+import static io.grpc.netty.shaded.io.netty.handler.codec.http.HttpResponseStatus.OK;
+import static io.grpc.netty.shaded.io.netty.handler.codec.http.HttpResponseStatus.UNAUTHORIZED;
+
+/**
+ * This is the http token server handler implementation.
+ */
+public class HttpTokenServerHandler extends SimpleChannelInboundHandler<HttpObject> {
+    private static TokenIssuer tokenIssuer;
+    private static boolean isAuthorized = false;
+    private static final Logger logger = LogManager.getLogger(HttpTokenServerHandler.class);
+
+    @Override
+    public void channelReadComplete(ChannelHandlerContext ctx) {
+        ctx.flush();
+    }
+
+    @Override
+    public void channelRead0(ChannelHandlerContext ctx, HttpObject msg) throws Exception {
+        FullHttpResponse response = null;
+
+        if (msg instanceof HttpRequest) {
+            HttpRequest req = (HttpRequest) msg;
+            boolean keepAlive = HttpUtil.isKeepAlive(req);
+
+            String authHeader = req.headers().get("Authorization");
+
+            if (authHeader == null) {
+                isAuthorized = false;
+                response = new DefaultFullHttpResponse(req.protocolVersion(), UNAUTHORIZED);
+                response.headers().set(CONTENT_TYPE, TEXT_PLAIN);
+                logger.error(APIConstants.StatusCodes.UNAUTHORIZED.getCode() +
```

Comment:
What is the error message sent to user in these scenarios

Reply:
Added error messages